backend/msgpack/queue_test.go

```diff
@@ -51,16 +51,19 @@ func TestInstanceQueueEnqueueQueueFull(t *testing.T) {
 	opts := testServerOptions().SetInstanceQueueSize(1)
 	queue := newInstanceQueue(testPlacementInstance, opts).(*queue)
 
-	// Fill up the queue and park the draining goroutine for 5 seconds
-	// so the queue remains full.
-	queue.writeFn = func([]byte) error { time.Sleep(5 * time.Second); return nil }
-	for i := 0; i < 10; i++ {
-		select {
-		case queue.bufCh <- msgpack.NewBufferedEncoder():
-		default:
-		}
+	// Fill up the queue and park the draining goroutine so the queue remains full.
+	parkCh := make(chan struct{})
+	drainCh := make(chan struct{})
+	queue.writeFn = func([]byte) error {
```

Comment:
If you do that, the main goroutine will be stuck at the second `queueCh <- ...` since the draining goroutine is blocked.